### PR TITLE
Artsy components

### PR DESCRIPTION
**Pull request by DJRHails**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a feature/bugfix branch** (right side). Don't request from master!
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Description
Please describe your pull request.
Screenshot if appropriate to improve evidence.

### Checklist
* [x] Have you done your changes on a separate branch. Branches MUST have descriptive names that start with either the `fix/` or `feature/` prefixes. Good examples are: `fix/signin-loop` or `feature/pr-templates`.
* [x] Have you got descriptive commit messages that would make sense if prefixed with "This commit will ..."
* [x] Will you squash when you merge this PR?

#### New Features

1. [x] Does your submission have a high coverage of your feature?
2. [x] Have you lint your code locally prior to submission?

#### Bugs

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests which cover this bug?
* [x] Have you successfully ran tests with your changes locally?

💔Thank you!

### CODE REVIEW

File: app/src/components/ProposalList.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  classes: any;
+}
+
+type ProposalListState = {
+  ids: any;
+}
+
+class ProposalList extends React.Component<ProposalListProps, ProposalListState> {
+  componentDidMount () {
+    this.props.drizzle.contracts.Governance.methods.getProposals().call()
```
Comment (Tooni) [inline app/src/components/ProposalList.tsx:19]: probably ought to split this into a variable and (e.g. governanceContract), and then calling the stuff on that variable

i also think when you do method chaining like this you should linebreak on each method, but idk if the linter allows it

File: app/src/components/Governance.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  drizzleState: any;
+}
+
+type GovernanceState = {
+  isGovernor: false;
+}
+
+class Governance extends React.Component<GovernanceProps, GovernanceState> {
+  componentDidMount () {
+    this.props.drizzle.contracts.Governance.methods.isGovernor(this.props.drizzleState.accounts[0]).call()
```
Comment (Tooni) [inline app/src/components/Governance.tsx:17]: extract out a variable. see other similar comment.

File: app/src/components/ProposalItem.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+type ProposalItemState = {
+  proposal: any;
+}
+
+class ProposalItem extends React.Component<ProposalItemProps, ProposalItemState> {
+  rejectProposal = () => {
+    console.log('Rejecting proposal ' + this.props.id);
+    this.props.drizzle.contracts.Governance.methods.reject(this.props.id).send({
```
Comment (Tooni) [inline app/src/components/ProposalItem.tsx:23]: extract out contract

File: app/src/components/ProposalItem.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  approveProposal = () => {
+    console.log('Approving proposal ' + this.props.id);
+    this.props.drizzle.contracts.Governance.methods.approve(this.props.id).send({
+      from: this.props.drizzleState.accounts[0],
+    });
+  }
+
+  componentDidMount () {
+    this.props.drizzle.contracts.ArtifactApplication.methods.getProposal(this.props.id).call()
```
Comment (Tooni) [inline app/src/components/ProposalItem.tsx]: extract out contract

File: app/src/components/ProposalItem.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    return (
+      <ListItem alignItems="flex-start" key={this.props.id}>
+        <Grid container direction="row">
+          <ArtworkInfo artwork={this.state.proposal} id={this.props.id}/>
+          <Button
+            variant="contained"
+            color="primary"
+            className={this.props.classes.approve}
+            onClick={(e) => this.approveProposal()}>
```
Comment (Tooni) [inline app/src/components/ProposalItem.tsx]: change to 

onClick={this.approveProposal}>

File: app/src/components/ProposalItem.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            Approve
+          </Button>
+          <Button
+            variant="contained"
+            color="secondary"
+            className={this.props.classes.reject}
+            onClick={(e) => this.rejectProposal()}>
```
Comment (Tooni) [inline app/src/components/ProposalItem.tsx]: change to 

onClick={this.rejectProposal}>

File: app/src/components/Register.tsx [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+  registerArtifact (event: any): void {
+    event.preventDefault();
+    const { drizzle, drizzleState } = this.props;
+
+    const currentAccount = drizzleState.accounts[0];
+    const artist = drizzleState.accounts[0]; // TODO: Update this to real artist's account
+    const imageUri = ''; // TODO: Implement image uploading
+
+    const stackId = drizzle.contracts.ArtifactApplication.methods.applyFor.cacheSend(
```
Comment (Tooni) [inline app/src/components/Register.tsx:56]: extract out contract

File: contracts/ArtifactApplication.sol [hunk 8]
```diff
@@ -22,8 +22,42 @@ contract ArtifactApplication {
     registry = _registry;
   }
 
-  function applyFor(address who, IArtifactRegistry.Artifact memory _artifact) public returns (uint) {
-    bytes memory data = abi.encodeWithSelector(registry.mint.selector, who, _artifact);
+  function applyFor(
+    address who,
```
Comment (Tooni) [inline contracts/ArtifactApplication.sol]: not sure 'who' is a good variable name. i do not know what it means

**Review by Tooni — COMMENTED** [review]
~8 compiler warnings about the 'any' type, which i guess is vestigial from the js conversion

File: app/src/components/ProposalList.tsx (hunk 1 above)
Comment (JamesDalboth) [reply app/src/components/ProposalList.tsx:19]: I agree about method chaing, as for splitting out into a seperate variable, I do not know enough about typescript or react to be able to work out how to do that, I've attempted it but I keep getting state errors so am going to ignore the contract variable comments

**Comment by JamesDalboth** [comment]
There is a lot of any type warnings, problem is more of those cases are for drizzle or drizzle contract which is a complicated type

**Review by Tooni — COMMENTED** [review]
Ignoring minor style changes, to merge

**Review by Tooni — APPROVED** [review]

**Pull request merged**